lightning/src/ln/channelmanager.rs

```diff
@@ -2382,6 +2399,14 @@ where
 
 			let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 			let peer_state = &mut *peer_state_lock;
+
+			if peer_state.has_pending_channel(&channel_id) {
```

Comment:
Two or possibly three more map look-ups in `force_close_channel_with_peer` given this will be either a pending inbound or outbound channel and the non-pending map is checked first. So total of five look-ups for the inbound case.

FWIW, if there was one map with an enum, a single look-up could give back the entry to use by `force_close_channel_with_peer`. Maybe macro could help in the current design?

Reply:
Yeah, single map with enum was considered. It's nice in cases like this, but I remember some friction elsewhere. Maybe something to consider again. I want to see how much worse it makes it in other areas again (and how big the commit would be to use an enum). Dropping the channel interface in the channel split PR probably makes it better now.

Reply:
Ok so it's a bit involved to take a look at using an enum in a single map, but there are some nice advantages. I'll just continue with fixing up the multi-map approach and luckily it's something that we can change internally to channel monitor in a future PR.